plugins/SegmentEditor/SegmentEditor.php

```diff
@@ -283,7 +285,20 @@ private function getSegmentIfIsUnprocessed()
             $storedSegment = null;
         }
 
-        return [$segment, $storedSegment, $isSegmentToPreprocess];
+
+        $lastArchiveData =  ArchiveInvalidator::getEarliestDateToRearchive();
```

Comment:
`$lastArchiveData` seems to be an improper variable name. Would be good to use names that make the code easier to understand.